### PR TITLE
doc: —pg flag

### PR DESCRIPTION
**Pull request by tianzhou**
The documentation for the --pg flag

### CODE REVIEW

**Comment by tianzhou** [comment]
@tisonkun

**Comment by tianzhou** [comment]
A draft for how user configures external postgres database via --pg. Will add the SSL section later.

This doc dictates from the user perspective. It still requires some implementation work to reach this user behavior.

**Review by d-bytebase — APPROVED** [review]

**Review by tisonkun — DISMISSED** [review]
LGTM. Although I have no idea about the change on `nuxt.config.js`

**Review by d-bytebase — APPROVED** [review]

**Comment by tianzhou** [comment]
> LGTM. Although I have no idea about the change on `nuxt.config.js`

My bad. Mixed some other branch this morning.

**Comment by tianzhou** [comment]
Ok, so since we agree on the user behavior, I will first coordinate @d-bytebase to make some internal cleanup before @tisonkun you can proceed with the remaining tasks.

File: docs/install/external-postgres.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   - CONNECT
+   - TEMPORARY
+   - EXECUTE
+   - USAGE
+
+## --pg connection string
+
+Supported format
+
+- _postgresql://user:secret@host:port_
```
Comment (d-bytebase) [inline docs/install/external-postgres.md:31]: We always require a dbname for external postgres support.

**Review by d-bytebase — APPROVED** [review]

File: docs/install/external-postgres.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Example:
+
+- postgresql://bytebase:z\*3kd2@example.com:5432
+- postgresql://bytebase:z\*3kd2@example.com:5432/meta
+- postgresql://bytebase:z\*3kd2@example.com:5432/meta?sslrootcert=root.pem
+
+### Notes
+
+- `user` must be specified.
+- `dbname` is optional. If not specified, we will follow the PostgreSQL convention to use the `user` name as the database name.
```
Comment (h3n4l) [inline docs/install/external-postgres.md:46]: Users must provide `dbname` or else [encounter the error](https://github.com/bytebase/bytebase/blob/93cb74b0118872ef810fc28e3e035f80aa809ef5/bin/server/cmd/root.go#L393).

**Comment by tianzhou** [comment]
@d-bytebase @h3n4l the postgres convention is to use the user name as the database name if no database name is specified. Do we intentionally not support that?

**Comment by h3n4l** [comment]
> 

@tianzhou From my point of view, I don't think we need to follow the Postgres conventions here. Instead, we should let the user explicitly declare which database to use. When using the `-pg` option, the user should be able to clearly know where the data in bytebase will be stored. I think this is friendly to users unfamiliar with Postgres, and it also won't bother those who are familiar with Postgres.

**Comment by tianzhou** [comment]
@h3n4l make sense. Updated. PTAL.

**Review by h3n4l — DISMISSED** [review]
LGTM.

**Comment by tianzhou** [comment]
The feature is released. Ready to go

File: docs/install/external-postgres.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+---
+title: Use External PostgreSQL Database
+order: 10300
+---
+
+# Use External PostgreSQL Database
+
+By default, Bytebase bundles an embedded PostgreSQL instance and stores its own metadata there. Alternatively, you can use supply [--pg](/docs/reference/command-line#--pg-string) to use an external PostgreSQL database to store those metadata.
```
Comment (d-bytebase) [inline docs/install/external-postgres.md:8]:
```suggestion
By default, Bytebase bundles an embedded PostgreSQL instance for storing its own metadata. Alternatively, you can supply [--pg](/docs/reference/command-line#--pg-string) to store these metadata in an external PostgreSQL database.
```

File: docs/install/external-postgres.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+   - TRIGGER
+   - CREATE
+   - CONNECT
+   - TEMPORARY
+   - EXECUTE
+   - USAGE
+
+## --pg connection string
+
+Supported format
```
Comment (d-bytebase) [inline docs/install/external-postgres.md:29]:
```suggestion
Supported format:
```

File: docs/install/external-postgres.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+- _postgresql://user:secret@host:port/dbname_
+
+- _postgresql://user:secret@host:port/dbname?sslrootcert=root.pem_
+
+Example:
+
+- postgresql://bytebase:z\*3kd2@example.com:5432/meta
```
Comment (d-bytebase) [inline docs/install/external-postgres.md:37]:
```suggestion
- _postgresql://bytebase:z\*3kd2@example.com:5432/meta
```

File: docs/install/external-postgres.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+- postgresql://bytebase:z\*3kd2@example.com:5432/meta?sslrootcert=root.pem
```
Comment (d-bytebase) [inline docs/install/external-postgres.md:38]:
```suggestion
- _postgresql://bytebase:z\*3kd2@example.com:5432/meta?sslrootcert=root.pem
```

**Review by d-bytebase — APPROVED** [review]

**Pull request merged**